sa/sa.go

```diff
@@ -355,7 +356,7 @@ func (ssa *SQLStorageAuthority) AddPrecertificate(ctx context.Context, req *sapb
 		isRenewal, err := ssa.checkFQDNSetExists(
 			ctx,
 			tx.SelectOne,
-			parsed.DNSNames)
+			identifier.SliceNewDNS(parsed.DNSNames))
```

Comment:
I think hard-coding these as being DNS identifiers here is the wrong move, and would be very easy to forget to fix in the future. Instead, we're going to need something like `identifier.FromCert()` which processes both the `.DNSNames` field _and_ the `.IPAddresses` field into the correct set of identifiers.